### ISSUE TITLE
Unable to install local rack on Linux (Ubuntu 18.04)

### ISSUE THREAD

**Issue by qnm** [convox/rack#2854]
I've update to the latest convox binary, but am unable to install a local rack:

```
qnm@glenlivet:~/github/app$ sudo convox update
Updating to 20180815173013... OK
```

```
qnm@glenlivet:~/github/app$ sudo convox rack install local
pulling: convox/rack:20180815173013
installing: /lib/systemd/system/convox.router.service
installing: /lib/systemd/system/convox.rack.convox.service
Authenticating with rack.convox... ERROR: invalid login
```

It looks like the service has started correctly:

```
qnm@glenlivet:~/github/app$ docker ps
CONTAINER ID        IMAGE                        COMMAND             CREATED             STATUS              PORTS                     NAMES
a251db4c89da        convox/rack:20180815173013   "bin/web"           3 minutes ago       Up 3 minutes        0.0.0.0:32768->5443/tcp   convox
```

```
qnm@glenlivet:~$ ps -aux | grep convox
root     31021  0.2  0.2 1009524 21328 ?       Ssl  22:37   0:01 /usr/local/bin/convox router
root     31100  0.0  0.2 702328 16616 ?        Ssl  22:37   0:00 /usr/local/bin/convox rack start --name convox
root     31141  0.0  0.3 871668 26332 ?        Sl   22:37   0:00 docker run --rm -e COMBINED=true -e COMBINED=true -e IMAGE=convox/rack:20180815173013 -e PROVIDER=local -e RACK=convox -e ROUTER=10.42.0.0 -e VERSION=20180815173013 -e VOLUME=/var/convox -i --label convox.rack=convox --label convox.type=rack -m 256m --name convox -p 5443 -v /var/convox:/var/convox -v /var/run/docker.sock:/var/run/docker.sock convox/rack:20180815173013
```

It looks like there's a rack

```
qnm@glenlivet:~$ convox racks
NAME                   STATUS 
digivizer/collection   running
digivizer/integration  running
digivizer/production   running
local/convox           running
```

But I'm unable to start
```
qnm@glenlivet:~/github/digivizer/social$ convox start
ERROR: Post https://rack.local/convox/apps?: dial tcp: lookup rack.local: no such host
```

I added an entry for `rack.local` to my hostfile but still:

```
qnm@glenlivet:~/github/digivizer/social$ convox start
ERROR: Post https://rack.local/convox/apps?: dial tcp 127.0.1.1:443: connect: connection refused
```

**Comment by qnm** [comment]
To add to this - I have seen the same problem on an OSX machine. More details when I get them.

**Comment by ddollar** [comment]
`rack.local` is not the correct hostname for the local rack so my guess is this is a local configuration issue. You could try clearing your `~/.convox` directory and re-authenticating.

**Comment by klaygomes** [comment]
Same problem here:

```
pulling: convox/rack:20180907231934
installing: /Library/LaunchDaemons/convox.router.plist
installing: /Library/LaunchDaemons/convox.rack.convox.plist
Authenticating with rack.convox... ERROR: invalid login
```

Then when I am trying to run `convox start`, I get:

```
convox start                                                                                                                                                                        
ERROR: Get https://rack.convox/system?: dial tcp: i/o timeout
```
After I deleted ~/.convox and authenticated again (convox login), I started getting:

```
convox start
ERROR: Post https://rack.local/convox/apps?: dial tcp: lookup rack.local: no such host
```

**Comment by qnm** [comment]
This seems to have improved - looks like I now successfully install a local rack, but I still can't use it.

```
qnm@glenlivet:~/github/app$ convox start
ERROR: local rack not found, try `sudo convox rack install local`

qnm@glenlivet:~/github/app$ sudo convox rack install local
[sudo] password for qnm: 
pulling: convox/rack:20180921222709
installing: /lib/systemd/system/convox.router.service
installing: /lib/systemd/system/convox.rack.convox.service
waiting for rack... OK

qnm@glenlivet:~/github/app$ convox racks
NAME                   STATUS 
digivizer/collection   running
digivizer/integration  running
digivizer/production   running
local/convox           running


qnm@glenlivet:~/github/app$ convox start
ERROR: Post https://rack.convox/apps?: dial tcp: lookup rack.convox on 127.0.0.53:53: no such host
```

**Comment by ddollar** [comment]
What do you have listening to DNS on 127.0.0.53?

**Comment by qnm** [comment]
Thanks for the prompt. I have it working now.

1. I had dnsmasq running (which appeared to relate to a previous convox install) which I removed and ensured that `systemd-resolved` was resolving for the system.
2. I ran the `convox rack start` command inside `convox.rack.convox.service` systemd init file and noted an error relating to my kernel. I updated `/etc/default/grub` with settings `GRUB_CMDLINE_LINUX="cgroup_enable=memory swapaccount=1"` and rebooted
3. I was able to start convox but it was painfully slow to start up. Checking the process list I note that `kubectl` was being executed. I removed `kubectl` from my system and `convox start` runs immediately.
